### PR TITLE
Clean up tests to make them compatible with dotty.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

File: test-suite/js/src/test/require-scala2/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala2.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  @Test def default_values_for_fields(): Unit = {
+    val obj = new DefaultFieldValues
+
+    /* Value class fields are initialized to null, instead of a boxed
+     * representation of the zero of their underlying types, as for a
+     * Scala class.
+     *
+     * Scala 3 correctly unboxes `null` as the zero of the underlying type of
+     * value classes, which means that this test "fails" in Scala 3, although
+     * it is in fact a progression.
```
Comment (gzm0) [inline test-suite/js/src/test/require-scala2/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala2.scala:34]: I find the usage of the term "progression" as an antonym to "regression" quite confusing (it wasn't clear to me what it meant in the commit message). If it is common usage though in dotty / scala land, then we should leave it.

File: test-suite/js/src/test/require-scala2/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala2.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+object NonNativeJSTypeTestScala2 {
+
+  class DefaultFieldValues extends js.Object {
+    var valueClass: SomeValueClass = _
+  }
+
+  class SomeValueClass(val i: Int) extends AnyVal
+
+}
```
Comment (gzm0) [inline test-suite/js/src/test/require-scala2/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala2.scala:49]: You should remove these things from `NonNativeJSTypeTest`. They are unused there (or if they still provide value to make sure value classes "work", comment on them).

File: test-suite/shared/src/test/require-scala2/org/scalajs/testsuite/compiler/WasPublicBeforeTyperTestScala2.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class WasPublicBeforeTyperTestScala2 {
+
+  /** This is a Scala 2.x-only test because it is a bug compatibility test,
+   *  anddotty does expose the bug.
```
Comment (gzm0) [inline test-suite/shared/src/test/require-scala2/org/scalajs/testsuite/compiler/WasPublicBeforeTyperTestScala2.scala:23]: typo: and_dotty

**Review by gzm0 — CHANGES_REQUESTED** [review]

File: test-suite/js/src/test/require-scala2/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala2.scala (hunk 1 above)
Comment (sjrd) [reply test-suite/js/src/test/require-scala2/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala2.scala:34]: Huh. I've seen that often in the scala/scala repo, and thought it was standard terminology. I can use "an improvement" instead, perhaps?

Comment (gzm0) [reply test-suite/js/src/test/require-scala2/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala2.scala:34]: "improvement" is definitely clear to me. But if progression is common, then maybe I just needed to learn :)

**Comment by sjrd** [comment]
Updated. I changed to "improvement", since I did not find any evidence that "progression" is used outside of scala/scala.

**Review by gzm0 — APPROVED** [review]

**Pull request merged**